spec/unit/provider/oneview_datacenter_synergy_spec.rb

```diff
@@ -0,0 +1,128 @@
+    let(:resource) do
+      Puppet::Type.type(:oneview_datacenter).new(
+        name: 'Datacenter',
+        ensure: 'present',
+        data:
+            {
+              'name' => 'Datacenter',
+              'width' => '5000',
+              'depth' => '5000'
+            },
+        provider: 'synergy'
+      )
+    end
+
+    let(:provider) { resource.provider }
+
+    let(:instance) { provider.class.instances.first }
+
+    it 'should be an instance of the provider Ruby' do
+      expect(provider).to be_an_instance_of Puppet::Type.type(:oneview_datacenter).provider(:synergy)
+    end
+
+    it 'should return that the resource does not exists' do
+      allow(resourcetype).to receive(:find_by).and_return([])
+      expect(provider.exists?).not_to be
+    end
+
+    it 'should create/add the datacenter' do
+      test = resourcetype.new(@client, resource['data'])
+      expect(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([])
+      expect(resourcetype).to receive(:find_by).with(anything, 'name' => resource['data']['name'])
+        .and_return([])
+      provider.exists?
+      allow_any_instance_of(resourcetype).to receive(:add).and_return(test)
+      expect(provider.create).to be
+    end
+  end
+
+  context 'given the min parameters' do
+    let(:resource) do
+      Puppet::Type.type(:oneview_datacenter).new(
+        name: 'Datacenter',
+        ensure: 'found',
+        provider: 'synergy'
+      )
+    end
+
+    let(:provider) { resource.provider }
+
+    let(:instance) { provider.class.instances.first }
+
+    it 'should not return any datacenters' do
+      allow(resourcetype).to receive(:find_by).and_return([])
+      expect(provider.exists?).not_to be
+      expect { provider.found }.to raise_error(/No Datacenter with the specified data were found on the Oneview Appliance/)
+    end
+  end
+
+  context 'given the min parameters' do
+    let(:resource) do
+      Puppet::Type.type(:oneview_datacenter).new(
+        name: 'Datacenter',
+        ensure: 'present',
+        data:
+            {
+              'name' => 'Datacenter'
+            },
+        provider: 'synergy'
+      )
+    end
+
+    let(:provider) { resource.provider }
+
+    let(:instance) { provider.class.instances.first }
+
+    before(:each) do
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      provider.exists?
+    end
+
+    it 'should return that the resource exists' do
+      expect(provider.exists?).to eq(true)
+    end
+
+    it 'should be able to get the visual content' do
+      visual_content = 'spec/support/fixtures/unit/provider/datacenter_visual_content.json'
+      allow_any_instance_of(resourcetype).to receive(:get_visual_content).and_return(File.read(visual_content))
+      expect(provider.get_visual_content).to eq(true)
+    end
+
+    it 'deletes the resource' do
```

Comment:
If I remove this line, I got this error:
1) Puppet::Type::Oneview_datacenter::ProviderSynergy given the min parameters deletes the resource
     Failure/Error: datacenter.map(&:remove)
     
     OneviewSDK::IncompleteResource:
       Please set uri attribute before interacting with this resource
     # ./lib/puppet/provider/oneview_datacenter/c7000.rb:37:in `map'
     # ./lib/puppet/provider/oneview_datacenter/c7000.rb:37:in `destroy'
     # ./spec/unit/provider/oneview_datacenter_synergy_spec.rb:113:in `block (3 levels) in <top (required)>'